chainer/functions/connection/n_step_lstm.py

```diff
@@ -0,0 +1,242 @@
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+if cuda.cudnn_enabled:
+    cudnn = cuda.cudnn
+    libcudnn = cuda.cudnn.cudnn
+
+
+class PointerArray(object):
+
+    def __init__(self, lst, back_pointer):
+        self._value = numpy.array(lst, dtype=numpy.intp)
+        # Store back_pointer to prevent the GC removes the original variable
+        self._back_pointer = back_pointer
+
+    @property
+    def data(self):
+        return self._value.ctypes.data
+
+
+def _make_tensor_descriptor_array(xs, rev=True):
+    """Make an array of pointers denoting pointers of tensor descriptors. 
+   """
+    descs = []
+    for x in xs:
+        if x.ndim < 3:
+            if rev:
+                shape = (1,) * (3 - x.ndim) + x.shape
+            else:
+                shape = x.shape + (1,) * (3 - x.ndim)
+            x = x.reshape(shape)
+        desc = cudnn.create_tensor_nd_descriptor(x, rev=True)
+        descs.append(desc)
+    return PointerArray([d.value for d in descs], descs)
+
+
+def _make_ptr_array(xs):
+    """Make an array of pointers denoting pointers of ndarrays.
+    """
+    return PointerArray([x.data.ptr for x in xs], xs)
+
+
+class NStepLSTM(function.Function):
+
+    def __init__(self, n_layers, train=True):
+        self.seed = 1337
+        self.dropout = 0.0
+        self.n_layers = n_layers
+        self.train = train
+
+    def check_type_forward(self, in_types):
+        h_type, c_type, x_type, w_type, b_type = in_types
+        type_check.expect(
+            h_type.dtype == numpy.float32,
+            c_type.dtype == numpy.float32,
+            x_type.dtype == numpy.float32,
+
+            h_type.ndim == 3,
+            h_type.shape[0] == self.n_layers,
+            c_type.ndim == 3,
+            c_type.shape[0] == self.n_layers,
+
+            # mini-batch size
+            h_type.shape[1] == c_type.shape[1],
+            h_type.shape[1] == x_type.shape[1],
+
+            # hidden size
+            h_type.shape[2] == c_type.shape[2],
+
+            w_type.ndim == 4,
+            w_type.shape[0] == self.n_layers,
+            w_type.shape[1] == 8,
+            h_type.shape[2] == w_type.shape[2],
+            h_type.shape[2] == w_type.shape[3],
+
+            b_type.ndim == 3,
+            b_type.shape[0] == self.n_layers,
+            b_type.shape[1] == 8,
+            h_type.shape[2] == b_type.shape[2],
+        )
+
+    def forward(self, inputs):
+        hx, cx, xs, ws, bs = inputs
+
+        length = xs.shape[0]
```

Comment:
I have a question here.
I'm assuming that the shape of `xs` is `(length, batch_size, the number of dimensions of the first LSTM layer)`.
But what is `length`?

`length` seems to be finally passed to `cudnnSetRNNDescriptor` as `seqLength` but there are no such arguments in the manual. (Actually,  cuDNN's manual tell me that `cudnnSetRNNDescriptor` has 8 arguments but `cudnn.h` defines 9 arguments. There seems difference between the manual and implementation. Do you know any documents about the meanings of `seqLength`?)